### PR TITLE
Introduce theme to third-party Popover component

### PR DESCRIPTION
**Pull request by ChaoLiangSuper**
Introduce theme to popover component
`theme` includes `dark/light/warn/error`

- [x] Test and coverage

## Description

Resolves https://github.com/Adslot/adslot-ui/issues/795

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Screenshots (if appropriate):
screen shot 2019-01-22 at 11 55 43 pm

### Should not affect other usage like:
screen shot 2019-01-23 at 10 29 12 am
 

## Check-list:

- [x] I have read the [Contributing](CONTRIBUTING.md) document.
- [x] I've thought about and labelled my PR/commit message appropriately.
- [ ] If this PR introduces breaking changes I've described the impact and migration path for existing applications.
- [x] CI is green (coverage, linting, tests).
- [x] I have updated the documentation accordingly.
- [x] I've two LGTMs/Approvals.
- [x] I've fixed or replied to all my code-review comments.
- [ ] I've manually tested with a buddy.
- [x] I've squashed my commits into one.

### CODE REVIEW

File: docs/examples/PopoverExample.jsx [hunk 1]
```diff
@@ -0,0 +1,121 @@
+import React from 'react';
+import Example from '../components/Example';
+import { Popover } from '../../src';
+
+class PopoverExample extends React.PureComponent {
+  render() {
+    return (
+      <>
```
Comment (adityasabnis) [inline docs/examples/PopoverExample.jsx]: React.Fragment?

File: src/components/third-party/Popover/index.jsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+export const themes = ['light', 'dark', 'warn', 'error'];
+
+class Popover extends React.Component {
+  static propTypes = {
+    id: PropTypes.string.isRequired,
+    theme: PropTypes.oneOf(themes),
+  };
+
+  render() {
+    const { theme, children, className, ...restProps } = this.props;
```
Comment (adityasabnis) [inline src/components/third-party/Popover/index.jsx]: why do we not have className in PropTypes.
and is it ok to just pass restProps to the component.
I thought all the props required, should have a known definition in propTypes
?

Comment (ChaoLiangSuper) [reply src/components/third-party/Popover/index.jsx]: Forget it I will add it.

File: docs/examples/HelpIconPopoverExample.jsx [hunk 3]
```diff
@@ -15,7 +15,7 @@ class HelpIconPopoverExample extends React.PureComponent {
 }
 
 const exampleProps = {
-  componentName: 'Help Icon',
+  componentName: 'Help Icon Popover',
```
Comment (ChaoLiangSuper) [inline docs/examples/HelpIconPopoverExample.jsx]: Fix broken link.

File: package-lock.json [hunk 4]
```diff
@@ -2579,7 +2579,7 @@
         },
         "string-width": {
           "version": "1.0.2",
-          "resolved": "http://registry.npmjs.org/string-width/-/string-width-1.0.2.tgz",
+          "resolved": "https://registry.npmjs.org/string-width/-/string-width-1.0.2.tgz",
```
Comment (lightbringer1991) [inline package-lock.json]: revert this file?

Comment (ChaoLiangSuper) [reply package-lock.json]: Why? should be https? 
Actually, last night I ran `npm i` at home. It almost changing everything...

File: src/styles/bootstrapOverrides/Popover.scss [hunk 5]
```diff
@@ -1,14 +1,176 @@
 @import '~styles/variable';
+@import '~styles/mixins/popoverTheme';
+@import '~bootstrap-sass/assets/stylesheets/bootstrap/mixins/reset-text';
 
 .popover {
```
Comment (veegandhi) [inline src/styles/bootstrapOverrides/Popover.scss]: @ChaoLiangSuper - can you not simple import ```~bootstrap-sass\assets\stylesheets\bootstrap\_popovers.scss``` and then override the required stuff?

**Pull request merged**